synapse/handlers/room_list.py

```diff
@@ -238,9 +238,12 @@ def handle_room(room_id):
                 addition *= -1
 
             try:
-                new_limit = sorted_rooms.index(last_room_id) + addition
-                if new_limit >= len(sorted_rooms):
+                if sorted_rooms.index(last_room_id) == len(sorted_rooms):
```

Comment:
actually, this looks like it should be `len() - 1`

actually, i've got confused again on how this is meant to work. This would be so much clearer with a few more comments......